app/utils/timing.server.ts

```diff
@@ -1,8 +1,25 @@
+import { type CreateReporter } from 'cachified'
+
 export type Timings = Record<
 	string,
 	Array<{ desc?: string; type: string; time: number }>
 >
 
+function createTimer(type: string, desc?: string) {
+	const start = performance.now()
+	return {
+		end(timings: Timings) {
+			let timingType = timings[type]
+
+			if (!timingType) {
+				// eslint-disable-next-line no-multi-assign
+				timingType = timings[type] = []
+			}
+			timingType.push({ desc, type, time: performance.now() - start })
+		},
+	}
+}
+
```

Comment:
not entirely happy with this abstraction. Needed something where the start and end could be called manually. Happy to take suggestions for other directions.

Reply:
Looks ok to me. I can't think of a better way.